MainWindow.cpp

```diff
@@ -49,7 +63,88 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), ui(new Ui::MainWi
 
 MainWindow::~MainWindow() { delete ui; }
 
-void MainWindow::closeEvent(QCloseEvent *event) { event->accept(); }
+void MainWindow::readSettings() {
+  QSettings settings;
+
+  settings.beginGroup("MainWindow");
+  restoreGeometry(settings.value("geometry").toByteArray());
+  restoreState(settings.value("state").toByteArray());
+  settings.endGroup();
+}
+
+void MainWindow::writeSettings() {
+  QSettings settings;
+
+  settings.beginGroup("MainWindow");
+  settings.setValue("geometry", saveGeometry());
+  settings.setValue("state", saveState());
+  settings.endGroup();
+}
+
+static inline QString recentFilesKey() { return QStringLiteral("recentFileList"); }
+static inline QString fileKey() { return QStringLiteral("file"); }
+
+bool MainWindow::hasRecentFiles() {
+  QSettings settings;
+
+  const int count = settings.beginReadArray(recentFilesKey());
+  settings.endArray();
+  return count > 0;
+}
+
+static QStringList readRecentFiles(QSettings &settings) {

```

Comment:
Fantastic example of something MainWindow.cpp doesn't need to be in charge of.

Reply:
Moved with the other logic for recent files to `Components/RecentFiles.cpp`